addons/website/static/src/builder/plugins/form/form_option_plugin.js

```diff
@@ -268,6 +267,15 @@ export class FormOptionPlugin extends Plugin {
             }));
         } else if (field.relation && field.relation !== "ir.attachment") {
             const fieldNames = field.fieldName ? [field.fieldName] : ["display_name"];
+            if (field.name === "state_id") {
+                // check if country_id is a field on the model
+                const modelFields = await this.services.orm.call(field.relation, "fields_get", [
+                    ["country_id"],
+                ]);
```

Comment:
I thing you can use this.authorizedFieldsCache. The rpc on "get_authorized_fields" do an "fields_get".